### PR TITLE
2-3-Compatibility of Python Code

### PR DESCRIPTION
**Pull request by vvolkl**
BEGINRELEASENOTES
- Fixes in python code for 2-3 compatibility

ENDRELEASENOTES

### CODE REVIEW

File: python/EventStore.py [hunk 1]
```diff
@@ -1,3 +1,6 @@
+
+from past.builtins import xrange
```
Comment (petricm) [inline python/EventStore.py]: I would rather not import the past to the future but the future to the past. So not give python 3 features from python 2, but rather assume it's python 2 and give it python 3 features. The code should read as Python 3 (so no visible xrange) so it's "easier" to drop python 2.

**Comment by petricm** [comment]
Please check the gihub actions for python, you still have several problems

File: python/EventStore.py (hunk 1 above)
Comment (vvolkl) [reply python/EventStore.py]: Good point. Done.

**Comment by petricm** [comment]
Also it would be good to install the python stuff under `lib[64]/python${Python_VERSION_MAJOR}.${Python_VERSION_MINOR}/site-packages`

**Comment by vvolkl** [comment]
> Also it would be good to install the python stuff under `lib[64]/python${Python_VERSION_MAJOR}.${Python_VERSION_MINOR}/site-packages`

Agreed, as there are actually a few issues to fix with the python code organization. I'd propose I do it in a separate PR.

File: python/podio_class_generator.py [hunk 2]
```diff
@@ -27,1083 +32,1107 @@
...
+    ostream_implementation += '  << std::endl;\n'
+    ostream_implementation = ostream_implementation.replace("{header_string}", ostream_header_string)
+
+    # ----------------------------------------------------------------------
+
+    refmembers = definition["OneToOneRelations"]
+    refvectors = definition["OneToManyRelations"]
+    vectormembers = definition["VectorMembers"]
+    nOfRefVectors = len(refvectors)
+    n
```
[the remark this answers is not in the document]
Comment (vvolkl) [reply python/podio_class_generator.py]: Interesting bug indeed. Something about the diff pattern matching?

File: setup.cfg [hunk 3]
```diff
@@ -21,6 +23,10 @@ ignore =
        E125
        # continuation line over-indented for visual indent
        E127
+       # line break after binary operator (contradicts W503) 
+       W504
+       # module level import not at top of file (poblem with ROOT)
+       E402
```
Comment (petricm) [inline setup.cfg]: E402 what if you just flag the lines of ROOT as false positives

Comment (vvolkl) [reply setup.cfg]: Done!

**Comment by vvolkl** [comment]
@petricm good point, I just switched everything to 4 spaces now.

**Comment by vvolkl** [comment]
As discussed in the EDM4hep meeting, I changed this to 2 spaces indent. @gaede should be good to merge now!

**Pull request merged**